src/libraries/Microsoft.PowerFx.Connectors/ConnectorFunction.cs

```diff
@@ -1037,9 +1038,124 @@ internal static ConnectorType GetCdpTableType(ICdpTableResolver tableResolver, s
             delegationInfo = ((DataSourceInfo)connectorType.FormulaType._type.AssociatedDataSources.First()).DelegationInfo;
             optionSets = symbolTable.OptionSets.Select(kvp => kvp.Value);
 
+            List<string> issues = new List<string>();
+            HashSet<string> fieldNames = new HashSet<string>(connectorType.Fields.Select(ct => ct.Name));            
+
+            // Table level validation
+            ValidatePropertyExist(serviceCapabilities?.FilterRestriction?.RequiredProperties, fieldNames, ref issues, "x-ms-capabilities/filterRestrictions/requiredProperties");
+            ValidatePropertyExist(serviceCapabilities?.FilterRestriction?.NonFilterableProperties, fieldNames, ref issues, "x-ms-capabilities/filterRestrictions/nonFilterableProperties");
+            ValidateFunctionExist(serviceCapabilities?.FilterSupportedFunctions, ref issues, "x-ms-capabilities/filterFunctionSupport");
+            ValidatePropertyExist(serviceCapabilities?.GroupRestriction?.UngroupableProperties, fieldNames, ref issues, "x-ms-capabilities/groupRestriction/ungroupableProperties");
+            ValidateOdataVersion(serviceCapabilities?.ODataVersion, ref issues, "x-ms-capabilities/oDataVersion");
+            ValidateServerPaging(serviceCapabilities?.PagingCapabilities?.ServerPagingOptions, ref issues, "x-ms-capabilities/serverPagingOptions");
+            ValidatePropertyExist(serviceCapabilities?.SortRestriction?.UnsortableProperties, fieldNames, ref issues, "x-ms-capabilities/sortRestrictions/unsortableProperties");
+            ValidatePropertyExist(serviceCapabilities?.SortRestriction?.AscendingOnlyProperties, fieldNames, ref issues, "x-ms-capabilities/sortRestrictions/ascendingOnlyProperties");
+           
+            // Column level validation
+            foreach (ConnectorType field in connectorType.Fields)
+            {
+                ColumnCapabilities columnCapabilities = field.Capabilities;
+                ValidateFunctionExist(columnCapabilities?.Capabilities?.FilterFunctions, ref issues, $"schema/item/properties/{Display(field.Name)}/x-ms-capabilities/filterFunctions");                
+            }
+
+            // We exclude salesforce here as we do not manage compound properties
+            // ex: in Accounts table, BillingAddress consists of (BillingStreet, BillingCity, BillingState, BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, BillingGeocodeAccuracy)
+            if (connectorName != "salesforce")
+            {
+                foreach (string err in issues)
+                {
+                    connectorType.AddError(err);
+                }
+            }
+
             return connectorType;
         }
 
+        private static void ValidatePropertyExist(IEnumerable<string> propertyList, HashSet<string> fields, ref List<string> issues, string location)        

```

Comment:
>HashSet [](http://example.com/codeflow?start=84&length=7)

nit : IReadOnlySet? (ie, to be clear we're not mutating the field list)

Reply:
Not with .Net Standard 2.0 if I'm correct (started w/ .Net 5)
It's a private method so the risk is very little
It's degrading perf by 3% (on .Net 9)
I can use ISet instead (with the same perf impact)